### ISSUE TITLE
Help needed in adding the "timestampz" alias for "timestamp with time zone"

### ISSUE THREAD

**Issue by bplower** [oguimbal/pg-mem#68]
Hey @oguimbal 👋  Thanks for making this and the pg-mem repos- they're awesome!

I'm working on adding pg-mem to a personal project and had a few migrations that used `timestampz` rather than the more verbose `timestamp with time zone`. I figured this was a good opportunity to give back to the project and add support for `timestampz`, but I'm afraid I'm a bit lost while trying to implement the type alias.

I'm not sure if I'm going in the right direction, but these are changes I've got so far

```diff
diff --git a/src/syntax/base.ne b/src/syntax/base.ne
index 486eb66..e504c04 100644
--- a/src/syntax/base.ne
+++ b/src/syntax/base.ne
@@ -147,6 +147,7 @@ kw_cache -> %word {% notReservedKw('cache')  %}
 kw_cascade -> %word {% notReservedKw('cascade')  %}
 kw_no -> %word {% notReservedKw('no')  %}
 kw_timestamp -> %word {% notReservedKw('timestamp')  %}
+kw_timestampz -> %word {% notReservedKw('timestampz')  %}
 kw_cycle -> %word {% notReservedKw('cycle')  %}
 kw_function -> %word {% notReservedKw('function')  %}
 kw_returns -> %word {% notReservedKw('returns')  %}
diff --git a/src/syntax/create-table.spec.ts b/src/syntax/create-table.spec.ts
index 9b28fe5..ad50d5c 100644
--- a/src/syntax/create-table.spec.ts
+++ b/src/syntax/create-table.spec.ts
@@ -165,6 +165,16 @@ describe('Create table', () => {
         }],
     });
 
+    checkCreateTable(['create table "test" (value timestampz)'], {
+        type: 'create table',
+        name: 'test',
+        columns: [{
+            kind: 'column',
+            name: 'value',
+            dataType: { name: 'timestamp with time zone' },
+        }],
+    });
+
     checkInvalid('create table"test"(value "timestamp" with time zone)');
     checkInvalid('create table"test"(value timestamp with "time" zone)');
 ```

With the keyword added, I'm seeing my new test failing, which makes sense considering some form of alias hasn't been defined:

```
  1) Create table
       parses create table "test" (value timestampz):

      Parser has not returned the expected AST
      + expected - actual

       {
         "columns": [
           {
             "dataType": {
      -        "name": "timestampz"
      +        "name": "timestamp with time zone"
             }
             "kind": "column"
             "name": "value"
           }
      
      at Context.<anonymous> (src/syntax/spec-utils.ts:76:26)
      at processImmediate (internal/timers.js:461:21)
```

However I am admittedly lost now 😅 I took a wild guess and added the word to the data_type_date definition, but this clearly isn't quite right, since it results in the "Ambiguous matches" test failing, saying there are 2 available options.

```diff
diff --git a/src/syntax/base.ne b/src/syntax/base.ne
index 486eb66..e504c04 100644
--- a/src/syntax/base.ne
+++ b/src/syntax/base.ne
@@ -247,6 +248,7 @@ data_type_text
 #https://www.postgresql.org/docs/9.5/datatype-datetime.html
 data_type_date
      ->  (%word {% anyKw('timestamp', 'time') %}) (%kw_with | %word {% kw('without') %}) (%word {% kw('time') %}) (%word {% kw('zone') %})
+     | (%word {% kw('timestampz') %})
     # | word {% kw('date') %}
     # | word {% kw('interval') %}
     # | word {% kw('timestamp') %} 
```

I don't have much experience with ASTs or with moo/nearley, but I think I can get this implemented with a little nudge in the right direction, so any help or info would be much appreciated. Thanks! 🙂

**Comment by bplower** [comment]
Thanks for the quick response Olivier! I think I've figured out what what we've overlooked here, and I think it's kinda funny 😅 

To start, I've confirmed that I have pg-mem version `1.7.6` installed. I went back and reproduced the original error that I was seeing. This line:
```sql
ALTER TABLE galaxy ADD COLUMN start_time timestamptz;
```

was causing this error:
```
     Error: type "timestamptz" does not exist

🐜 This seems to be an execution error, which means that your request syntax seems okay,
but the resulting statement cannot be executed → Probably not a pg-mem error.

*️⃣ Reconsituted failed SQL statement: ALTER TABLE "galaxy"  ADD COLUMN "start_time" "timestamptz"

👉 You can file an issue at https://github.com/oguimbal/pg-mem along with a way to reproduce this error (if you can), and  the stacktrace:
```

(I also love your error reporting!! I aspire to write code of this quality 🤩 )

I was working on this at like 2-3am my time, so I was super tired and totally overlooked that the keyword is "timestamp**tz**", not "timestamp**z**". So I opened the issue and omitted the "T" in the keyword and local testing. Since the bug itself is that the "T" is missing in the keyword, it wasn't immediately apparent what was wrong while reading your response.

So the root issue here is that the type synonym is implemented as `timestampz`, when that is not a valid sql datatype, and it should instead be `timestamptz`. I feel like I've made this same exact a million times before 😂 That should be a super easy fix though! I can open a PR for this later this evening if you're cool with that 👍

**Comment by oguimbal** [comment]
Ah ! A typo... nice catch :)

Yes of course, I'm cool with that !

(that said, I'm affraid fixing the still typo wont change the fact that pg-mem is not timezone-aware... `timestamp` and `timestamptz` will behave identitically)

**Comment by bplower** [comment]
> wont change the fact that pg-mem is not timezone-aware

Oh right! I completely lost track of that part. 

`timestamp` and `timestamptz` working the same is totally fine for my use case, but I could see that being an issue for other prospective users 🤔  

I'll play around with the repo locally- I'll remove the type synonym and see how far I get in implementing support for time zones. I'm sure I'll have more questions about all that, but I can ask those in another issue to avoid cluttering this one.

**Comment by oguimbal** [comment]
Fixed with ` pg-mem@1.7.7`
